fillers/eips/eip4844/util_blobhash.py

```diff
@@ -0,0 +1,300 @@
+                        {cls.address("blobhash_return")}, 0, 0, 32, 0, 32))
+                    let blobhash := mload(0)
+                    sstore(end, blobhash)
+                    return(0, 0)
+                }}
+                """
+            ),
+            "staticcall": Yul(
+                f"""
+                {{
+                    let pos := calldataload(0)
+                    let end := calldataload(32)
+                    for {{ }} lt(pos, end) {{ pos := add(pos, 1) }}
+                    {{
+                    mstore(0, pos)
+                    pop(staticcall(gas(),
+                        {cls.address("blobhash_return")}, 0, 32, 0, 32))
+                    let blobhash := mload(0)
+                    sstore(pos, blobhash)
+                    }}
+
+                    mstore(0, end)
+                    pop(staticcall(gas(),
+                        {cls.address("blobhash_return")}, 0, 32, 0, 32))
+                    let blobhash := mload(0)
+                    sstore(end, blobhash)
+                    return(0, 0)
+                }}
+                """
+            ),
+            "create": Yul(
+                """
+                {
+                    calldatacopy(0, 0, calldatasize())
+                    pop(create(0, 0, calldatasize()))
+                }
+                """
+            ),
+            "create2": Yul(
+                """
+                {
+                    calldatacopy(0, 0, calldatasize())
+                    pop(create2(0, 0, calldatasize(), 0))
+                }
+                """
+            ),
+            "initcode": Yul(
+                f"""
+                {{
+                   for {{ let pos := 0 }} lt(pos, 10) {{ pos := add(pos, 1) }}
+                   {{
+                    let blobhash := {blobhash_verbatim}
+                        (hex"{Op.BLOBHASH.hex()}", pos)
+                    sstore(pos, blobhash)
+                   }}
+                   return(0, 0)
+                }}
+                """
+            ),
+        }
+        code = code.get(context_name)
+        if code is None:
+            raise ValueError(f"Invalid context: {context_name}")
+        return code
+
+    @classmethod
+    def created_contract(cls, context_name):
+        """
+        Maps contract creation to a specific context to a specific address.
+        """
+        contract = {
+            "tx_created_contract": compute_create_address(TestAddress, 0),
+            "create": compute_create_address(
+                cls.address("create"),
+                0,
+            ),
+            "create2": compute_create2_address(
+                cls.address("create2"),
+                0,
+                cls.code("initcode").assemble(),
+            ),
+        }
+        contract = contract.get(context_name)
+        if contract is None:
+            raise ValueError(f"Invalid contract: {context_name}")
+        return contract
+
+
+class BlobhashScenario:
```

Comment:
I really don't see why this needs to be a class, both methods can simply be two independent functions.

Reply:
Was personal preference here, I thought it would be cleaner so we can just import the class instead of multiple functions.

Happy to change them all to functions? But maybe classes would be nice if we use a single utility file for all the 4844 tests in the future, i.e so we can easily find out what functions belong to each test module

Reply:
Ah yes, that seems alright then :)